src/main/java/ui/GuiElement.java

```diff
@@ -0,0 +1,61 @@
+package ui;
+
+import backend.resource.TurboIssue;
+import backend.resource.TurboLabel;
+import backend.resource.TurboMilestone;
+import backend.resource.TurboUser;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class GuiElement {
+    private final TurboIssue issue;
+    private final List<TurboLabel> labels;
+    private final Optional<TurboMilestone> milestone;
+    private final Optional<TurboUser> assignee;
+
+    public GuiElement(TurboIssue issue,
+                      List<TurboLabel> labels,
+                      Optional<TurboMilestone> milestone,
+                      Optional<TurboUser> assignee) {
+        this.issue = issue;
+        this.labels = labels;
+        this.milestone = milestone;
+        this.assignee = assignee;
+    }
+
+    public TurboIssue getIssue() {
+        return issue;
+    }
+
+    /**
+     * Returns all labels related to an issue (including labels that have been added and then removed).
+     * Use getLabelsOfIssue to retrieve an issue's current labels instead.
+     *
+     * @return Labels related to an issue, or null if the GuiElement has been constructed as such.
+     */
+    public List<TurboLabel> getLabels() {
```

Comment:
I was actually thinking of splitting the current labels and the removed labels lists into two lists, since we do need the list of current labels to display as `issueDetails`. What do you think?